route_builder.go

```diff
@@ -24,6 +24,7 @@ type RouteBuilder struct {
 	httpMethod  string        // required
 	function    RouteFunction // required
 	filters     []FilterFunction
+	conditions []RouteSelectionConditionFunction
```

Comment:
I renamed this from `iffunctions` to `conditions` as that seemed to read better.